ejb3/src/main/java/org/jboss/as/ejb3/component/stateful/StatefulSessionBeanImmutability.java

```diff
@@ -5,27 +5,43 @@
 
 package org.jboss.as.ejb3.component.stateful;
 
+import java.lang.reflect.Field;
 import java.lang.reflect.InvocationHandler;
 
 import org.jboss.as.ee.component.ProxyInvocationHandler;
 import org.jboss.invocation.proxy.ProxyFactory;
 import org.kohsuke.MetaInfServices;
 import org.wildfly.clustering.ee.Immutability;
+import org.wildfly.security.ParametricPrivilegedAction;
+import org.wildfly.security.manager.WildFlySecurityManager;
 
 /**
  * Immutability test for EJB proxies, whose serializable placeholders are immutable.
  * @author Paul Ferraro
  */
 @MetaInfServices(Immutability.class)
 public class StatefulSessionBeanImmutability implements Immutability {
+    private static final ParametricPrivilegedAction<InvocationHandler, Object> GET_INVOCATION_HANDLER = new ParametricPrivilegedAction<>() {
+        @Override
+        public InvocationHandler run(Object object) {
+            // Since there is a low probability that this object is actually a SFSB proxy, avoid Class.getDeclaredField(...) which will typically throw/catch a NoSuchFieldException
+            for (Field field : object.getClass().getDeclaredFields()) {
+                if (field.getName().equals(ProxyFactory.INVOCATION_HANDLER_FIELD)) {
+                    field.setAccessible(true);
```

Comment:
We should probably use `field.trySetAccessible()` here since `setAccessible(boolean)` is deprecated.

Reply:
Hmm... That's news to me - when was AccessableObject.setAccessible(...) deprecated?

Reply:
Well, it's not. I'm not sure where I got that idea from. Sorry about that.

Reply:
Oh, maybe. I guess I thought it was because of modules and the preference towards `VarHandle` now. I do know `isAccessible()` is [deprecated](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/reflect/AccessibleObject.html#isAccessible()).